src/firefly/js/charts/ui/PlotlyWrapper.jsx

```diff
@@ -236,10 +243,10 @@ const now = Date.now();
                         Plotly.Plots.resize(this.div);
                         break;
                     case RenderType.UPDATE:
-                        Plotly.update(this.div, data, layout);
+                        Plotly.update(this.div, data, cloneDeep(layout));
```

Comment:
the layout object gets changed after calling Plotly function for bar chart case and that make the logic confliction in submitChanges of BasicOptions. This happens while calling Plotly.newPlot.  We could talk in more detail about this solution.

Reply:
Cindy added  the `cloneDeep` because plotly was adding properties to the layout object.  It does not treat it as immutable data.  If this is a problem the the `optimize` then we have a difficult problem. Who can tell us for sure that it is a problem with `optimize`?